### PR TITLE
docs: add RELEASING.md and TESTING.md

### PR DESCRIPTION
**Pull request by Haprog**
Add documentation for how to create a new release in GitHub and npm (step by step).
Add documentation for how to test the package using `npm link` and `npm pack`.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+```shell
+npm login
+```
+
+## 1. Update the version in package.json
+
+> Currently pushing directly to `main` branch is not allowed due to branch protection rules so you can't directly push the commit generated by `npm version` into `main`. So we need to first create a release branch for getting the `package.json` version update merged via a pull request.
+>
+> If you have permissions to edit branch protection rules in GH, this can be done simpler by avoidin a separate PR and review/approval just for version bump commit if you temporarily remove the PR requirement for changes to `main`.
```
Comment (ninopenttinen) [inline RELEASING.md:17]: I would rather advice to merge the changes via PR, but to skip the review/approval process by merging with admin-rights. (My guess is that editing the branch protection rules requires admin-rights anyways, and if not, **it should**).

**Review by ninopenttinen — CHANGES_REQUESTED** [review]
Very good instructions, greatly appriciate this 👍 One request before merging though.

File: RELEASING.md (hunk 1 above)
Comment (Haprog) [reply RELEASING.md:17]: Good point. I'll change this part a bit.

There are also many other ways releasing could be configured to be a bit more automatic but I think this is good enough for now with just the instructions (after I make a small update). For example version bump commit and/or release could be done automatically by GitHub actions when publishing a "GitHub Release" or when merging to a separate release branch, but that would require more configuration and thinking of process.

Comment (ninopenttinen) [reply RELEASING.md:17]: Yep totally agree.

Comment (Haprog) [reply RELEASING.md]: I pushed an update.

**Review by ninopenttinen — APPROVED** [review]

**Pull request merged**